plugin/storage/factory.go

```diff
@@ -117,10 +118,16 @@ func (f *Factory) CreateSpanWriter() (spanstore.Writer, error) {
 		}
 		writers = append(writers, writer)
 	}
+	var spanWriter spanstore.Writer
 	if len(f.SpanWriterTypes) == 1 {
-		return writers[0], nil
+		spanWriter = writers[0]
+	} else {
+		spanWriter = spanstore.NewCompositeWriter(writers...)
 	}
-	return spanstore.NewCompositeWriter(writers...), nil
+	return spanstore.NewDownSamplingWriter(spanWriter, spanstore.DownSamplingOptions{
```

Comment:
why do we always apply downsampler? if f.DownSamplingRatio == 1 it only creates unnecessary overhead. Rather than checking for the same condition there _on every span_, we can check it here _once_ and skip the downsampler.

Reply:
i thought we discussed about applying downsampling dynamically in the future, so we should always apply it?

Reply:
Maybe we can add it in the future then. Right now we don't have any dynamic control plane to benefit from such setup.

Reply:
Sounds good.